testing/endtoend/evaluators/beaconapi_evaluators/beacon_api.go

```diff
@@ -307,150 +497,145 @@ func withCompareBeaconAPIs(beaconNodeIdx int) error {
 				if err != nil {
 					return err
 				}
-				beaconPathsAndObjects[path].prysmResps[key] = prysmr
-				beaconPathsAndObjects[path].lighthouseResps[key] = lighthouser
+				requests[path].prysmResps[key] = prysmr
+				requests[path].lighthouseResps[key] = lighthouser
 			default:
 				return fmt.Errorf("unknown encoding type %s", key)
 			}
 		}
 	}
-	return orderedEvaluationOnResponses(beaconPathsAndObjects, genesisResp)
+	return postEvaluation(beaconNodeIdx, requests)
 }
 
-func orderedEvaluationOnResponses(beaconPathsAndObjects map[string]metadata, genesisData *beacon.GetGenesisResponse) error {
-	forkPathData := beaconPathsAndObjects["/beacon/states/{param1}/fork"]
-	prysmForkData, ok := forkPathData.prysmResps["json"].(*beacon.GetStateForkResponse)
+// postEvaluation performs additional evaluation after all requests have been completed.
+// It is useful for things such as checking if specific fields match between endpoints.
+func postEvaluation(beaconNodeIdx int, requests map[string]metadata) error {
+	// verify that SSZ responses have the correct structure
+	forkData := requests["/beacon/states/{param1}/fork"]
+	fork, ok := forkData.prysmResps["json"].(*beacon.GetStateForkResponse)
 	if !ok {
-		return errors.New("failed to cast type")
+		return errJsonCast
 	}
-	lighthouseForkData, ok := forkPathData.lighthouseResps["json"].(*beacon.GetStateForkResponse)
-	if !ok {
-		return errors.New("failed to cast type")
-	}
-	if prysmForkData.Data.Epoch != lighthouseForkData.Data.Epoch {
-		return fmt.Errorf("prysm epoch %v does not match lighthouse epoch %v",
-			prysmForkData.Data.Epoch,
-			lighthouseForkData.Data.Epoch)
-	}
-
-	finalizedEpoch, err := strconv.ParseUint(prysmForkData.Data.Epoch, 10, 64)
+	finalizedEpoch, err := strconv.ParseUint(fork.Data.Epoch, 10, 64)
 	if err != nil {
 		return err
 	}
-	blockPathData := beaconPathsAndObjects["/beacon/blocks/{param1}"]
-	sszrspL, ok := blockPathData.prysmResps["ssz"].([]byte)
-	if !ok {
-		return errors.New("failed to cast type")
-	}
-	sszrspP, ok := blockPathData.lighthouseResps["ssz"].([]byte)
+	blockData := requests["/beacon/blocks/{param1}"]
+	ssz, ok := blockData.prysmResps["ssz"].([]byte)
 	if !ok {
-		return errors.New("failed to cast type")
+		return errSszCast
 	}
 	if finalizedEpoch < helpers.AltairE2EForkEpoch+2 {
-		blockP := &ethpb.SignedBeaconBlock{}
-		blockL := &ethpb.SignedBeaconBlock{}
-		if err := blockL.UnmarshalSSZ(sszrspL); err != nil {
-			return errors.Wrap(err, "failed to unmarshal lighthouse ssz")
-		}
```

Comment:
I removed the Lighthouse part because:
- we compare raw bytes by default in the evaluator, so if one response unmarshals, the other will too
- if lighthouse's response does not unmarshal, then it's a lighthouse bug